pageserver/src/tenant.rs

```diff
@@ -1620,8 +1623,15 @@ impl Tenant {
                         .filter(|timeline| timeline.current_state() != TimelineState::Broken);
 
                     // Spawn gc and compaction loops. The loops will shut themselves
-                    // down when they notice that the tenant is inactive.
-                    tasks::start_background_loops(self);
+                    // down in response to the cancellation token getting dropped.
+                    let background_loops_cancel = CancellationToken::new();
+                    let existing = self
+                        .background_loops_cancel
+                        .lock()
+                        .unwrap()
+                        .replace(background_loops_cancel.clone());
+                    assert!(existing.is_none(), "we don't support re-activation");
+                    tasks::start_background_loops(self, background_loops_cancel.clone());
```

Comment:
If we later fail to activate the timelines, we need to cancel the background loops. We're not donig that currently.

Reply:
This worry goes away with https://github.com/neondatabase/neon/pull/4319 which makes the activation infallibe.